offload_vio/server_tx/plugin.cpp

```diff
@@ -15,106 +18,145 @@ class server_writer : public plugin {
-        imu_params->set_acc_noise(imu_int_input->params.acc_noise);
-        imu_params->set_gyro_walk(imu_int_input->params.gyro_walk);
-        imu_params->set_acc_walk(imu_int_input->params.acc_walk);
-        vio_output_proto::Vec3* n_gravity = new vio_output_proto::Vec3();
-        n_gravity->set_x(imu_int_input->params.n_gravity.x());
-        n_gravity->set_y(imu_int_input->params.n_gravity.y());
-        n_gravity->set_z(imu_int_input->params.n_gravity.z());
-        imu_params->set_allocated_n_gravity(n_gravity);
-        imu_params->set_imu_integration_sigma(imu_int_input->params.imu_integration_sigma);
-        imu_params->set_nominal_rate(imu_int_input->params.nominal_rate);
-        protobuf_imu_int_input->set_allocated_imu_params(imu_params);
-
-        vio_output_proto::Vec3* biasAcc = new vio_output_proto::Vec3();
-        biasAcc->set_x(imu_int_input->biasAcc.x());
-        biasAcc->set_y(imu_int_input->biasAcc.y());
-        biasAcc->set_z(imu_int_input->biasAcc.z());
-        protobuf_imu_int_input->set_allocated_biasacc(biasAcc);
-
-        vio_output_proto::Vec3* biasGyro = new vio_output_proto::Vec3();
-        biasGyro->set_x(imu_int_input->biasGyro.x());
-        biasGyro->set_y(imu_int_input->biasGyro.y());
-        biasGyro->set_z(imu_int_input->biasGyro.z());
-        protobuf_imu_int_input->set_allocated_biasgyro(biasGyro);
-
-        vio_output_proto::Vec3* position_int = new vio_output_proto::Vec3();
-        position_int->set_x(imu_int_input->position.x());
-        position_int->set_y(imu_int_input->position.y());
-        position_int->set_z(imu_int_input->position.z());
-        protobuf_imu_int_input->set_allocated_position(position_int);
-
-        vio_output_proto::Vec3* velocity = new vio_output_proto::Vec3();
-        velocity->set_x(imu_int_input->velocity.x());
-        velocity->set_y(imu_int_input->velocity.y());
-        velocity->set_z(imu_int_input->velocity.z());
-        protobuf_imu_int_input->set_allocated_velocity(velocity);
-
-        vio_output_proto::Quat* rotation_int = new vio_output_proto::Quat();
-        rotation_int->set_w(imu_int_input->quat.w());
-        rotation_int->set_x(imu_int_input->quat.x());
-        rotation_int->set_y(imu_int_input->quat.y());
-        rotation_int->set_z(imu_int_input->quat.z());
-        protobuf_imu_int_input->set_allocated_rotation(rotation_int);
-
-        vio_output_proto::VIOOutput* vio_output_params = new vio_output_proto::VIOOutput();
-        vio_output_params->set_allocated_slow_pose(protobuf_slow_pose);
-        vio_output_params->set_allocated_imu_int_input(protobuf_imu_int_input);
-
-        publisher.Send(*vio_output_params);
-        delete vio_output_params;
+        // Check if a socket connection has been established
+        if (write_socket != NULL) {
+            // Construct slow pose for output
+            vio_output_proto::SlowPose* protobuf_slow_pose = new vio_output_proto::SlowPose();
```

Comment:
Are these deleted after allocation, I only see delete vio_output_params

Reply:
`protobuf_slow_pose` is owned by `vio_output_params` by using the `set_allocated_xxx` method. So deleting `vio_output_params` should also delete `protobuf_slow_pose` [Reference](https://protobuf.narkive.com/e7JbDIMe/whose-responsibility-to-delete-the-pb-message-which-is-added-by-set-allocated-xxx-xxx-ptr).  
Also, according to [this post](https://stackoverflow.com/questions/33960999/protobuf-will-set-allocated-delete-the-allocated-object), protobuf takes care of deleting the object when it goes out of scope. So if I understand correctly, we don't even need to delete `vio_output_params`. But no harm to keep it this way.

Reply:
ok